src/node_worker.cc

```diff
@@ -192,13 +192,8 @@ class WorkerThreadData {
         *static_cast<bool*>(data) = true;
       }, &platform_finished);
 
-      // The order of these calls is important; if the Isolate is first disposed
-      // and then unregistered, there is a race condition window in which no
-      // new Isolate at the same address can successfully be registered with
-      // the platform.
-      // (Refs: https://github.com/nodejs/node/issues/30846)
-      w_->platform_->UnregisterIsolate(isolate);
       isolate->Dispose();
+      w_->platform_->UnregisterIsolate(isolate);
```

Comment:
The comment here explains why undoing this change introduces a bug.

Please don’t do that.

For working around #31752, it should be sufficient to only reverse the changes made to `node_main_instance.cc` only.

Reply:
Also, the release information is wrong here – #31752 **does** reproduce on v13.x and master.